pkg/controllers/management/template/controller.go

```diff
@@ -0,0 +1,262 @@
+		projectLister:         management.Management.Projects("").Controller().Lister(),
+	}
+
+	management.Management.Catalogs("").Controller().AddHandler("createGlobalTemplateBinding", t.templateBindingForGlobalCatalog)
+	management.Management.ClusterCatalogs("").Controller().AddHandler("createClusterTemplateBinding", t.templateBindingForClusterCatalog)
+	management.Management.ProjectCatalogs("").Controller().AddHandler("createProjectTemplateBinding", t.templateBindingForProjectCatalog)
+	management.Management.ClusterRoleTemplateBindings("").Controller().AddHandler("addUserToClusterCatalog", t.updateCRBSubjectListForClusterCatalogTemplates)
+	management.Management.ProjectRoleTemplateBindings("").Controller().AddHandler("addUserToProjectCatalog", t.updateCRBSubjectListForProjectCatalogTemplates)
+}
+
+func buildSubjectFromRTB(binding interface{}) (v1.Subject, bool, error) {
+	var userName, groupPrincipalName, groupName, name, kind string
+	if rtb, ok := binding.(*v3.ProjectRoleTemplateBinding); ok {
+		userName = rtb.UserName
+		groupPrincipalName = rtb.GroupPrincipalName
+		groupName = rtb.GroupName
+		// only create subjects to add to the projectcatalog's templates bindings, if the roletemplate on the current rtb should have access to project catalogs
+		if !roleValidForProjectCatalogTemplates(rtb.RoleTemplateName) {
+			return v1.Subject{}, false, nil
+		}
+	} else if rtb, ok := binding.(*v3.ClusterRoleTemplateBinding); ok {
+		userName = rtb.UserName
+		groupPrincipalName = rtb.GroupPrincipalName
+		groupName = rtb.GroupName
+		// only create subjects to add to the clustercatalog's templates bindings, if the roletemplate on the current rtb should have access to cluster catalogs
+		if !roleValidForClusterCatalogTemplates(rtb.RoleTemplateName) {
+			return v1.Subject{}, false, nil
+		}
+	} else {
+		return v1.Subject{}, false, fmt.Errorf("unrecognized roleTemplateBinding type: %v", binding)
+	}
+
+	if userName != "" {
+		name = userName
+		kind = "User"
+	}
+
+	if groupPrincipalName != "" {
+		if name != "" {
+			return v1.Subject{}, false, fmt.Errorf("roletemplatebinding has more than one subject fields set: %v", binding)
+		}
+		name = groupPrincipalName
+		kind = "Group"
+	}
+
+	if groupName != "" {
+		if name != "" {
+			return v1.Subject{}, false, fmt.Errorf("roletemplatebinding has more than one subject fields set: %v", binding)
+		}
+		name = groupName
+		kind = "Group"
+	}
+
+	if name == "" {
+		return v1.Subject{}, false, fmt.Errorf("roletemplatebinding doesn't have any subject fields set: %v", binding)
+	}
+
+	return v1.Subject{
+		Kind: kind,
+		Name: name,
+	}, true, nil
+}
+
+func (tm *RBACTemplateManager) updateCRBSubjects(crbName string, removeUser bool, reconciledSubjects []v1.Subject, currentSubject v1.Subject) error {
+	updated := false
+	crb, err := tm.crbLister.Get("", crbName)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+	if crb == nil {
+		return nil
```

Comment:
it feels like it would make sense to handle crb create and update in one method. Otherwise all sorts of race conditions are possible. The logic should be as follows:

* fetch crb
* if not found, create. If create raises an error, check if error is already exists. If it is, proceed to update.